src/GRA.Domain.Service/AvatarService.cs

```diff
@@ -145,6 +145,47 @@ public async Task<ICollection<AvatarLayer>> GetUserWardrobeAsync()
             return layers.Where(_ => _.AvatarItems.Count > 0).ToList();
         }
 
+        public async Task<List<List<AvatarLayer>>> GetWardrobe(List<int> itemIds)
+        {
+            var siteId = GetCurrentSiteId();
+            var layers = await GetLayersAsync();
+            var layerGroupings = layers
+                .GroupBy(_ => _.GroupId)
+                .Select(_ => _.ToList())
+                .ToList();
+            var filePath = _pathResolver.ResolveContentPath($"site{siteId}/avatars/");
+            if (itemIds != null)
+            {
+                var items = await GetItemsByIdsAsync(itemIds);
+                foreach (var layer in layerGroupings.SelectMany(_ => _))
+                {
+                    var item = items
+                        .FirstOrDefault(_ => _.AvatarLayerId == layer.Id);
+                    if (item != null)
+                    {
+                        var fileName = "item";
+                        var element =
```

Comment:
A check can be done on the layer if it has ShowColorSelector set to true to see if a color needs to be added.

Reply:
Resolved in commit 91763bd